### PR TITLE
Updated the DNN Platform project details

### PR DESCRIPTION
**Pull request by hismightiness**
Updates include:  

* Fixed typos  
* Updated links for the project location  
* Replaced the Persona Bar repo location with the DNN Community repos  
* Fixed the project broken image  
* Added new references to community websites and resources, including in GitHub and our open-source Forge  
* Updated existing links to newer resources on the official community website  
* Added more details about the project features and longevity  
* Added trailing white space on all lines for maximum MD compatibility

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet-foundation/website?pullRequest=224) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet-foundation/website?pullRequest=224) before we can accept your contribution.<br/><br/>:x: hismightiness [sign now](https://cla.dotnetfoundation.org/dotnet-foundation/website?pullRequest=224)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet-foundation/website?pullRequest=224) it.</sub>

File: input/projects/data/dnn-platform.md [hunk 1]
```diff
@@ -2,48 +2,50 @@
...
+
+* [Contribute](https://github.com/dnnsoftware/Dnn.Platform/blob/development/CONTRIBUTING.md)  
+* [Community Blogs](https://dnncommunity.org/blogs)  
+* [Documentation](https://dnndocs.com/)  
+* [Extension Library (a.k.a Forge, for open-source extensions only)](https://dnncommunity.org/Forge)  
+* [DNN on NuGet](https://www.nuget.org/profiles/dotnetnuke)  
+* [Forums](https://dnncommunity.org/forums)  
+* [Showcase](https://dnncommunity.org/Showcase)  
+* [DNN on Facebook](https://www.facebook.com/groups/dnnconnect)  
+* [DNN on Slack](https://www.facebook.com/groups/dnnconnect)
```
Comment (david-poindexter) [inline input/projects/data/dnn-platform.md:49]:
```suggestion
* [DNN on Slack](https://openhelp.slack.com)  
```

**Comment by hismightiness** [comment]
Thanks, David.  Nice catch!  :D

**Comment by david-poindexter** [comment]
> Thanks, David. Nice catch! :D

You are most welcome sir - thanks for the contribution!  This all needed to be updated for sure.  😉

**Pull request merged**